pkg/executor/sortexec/sort.go

```diff
@@ -272,9 +258,13 @@ func (e *SortExec) InitInParallelModeForTest() {
 */
 func (e *SortExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	if e.fetched.CompareAndSwap(false, true) {
-		e.initCompareFuncs()
+		err := e.initCompareFuncs()
```

Comment:
I think you should remember this error in `SortExec`, so if `initCompareFuncs` fails, every call of `Next` will return error, otherwise, the second call of `Next` will go to L266 directly?

Reply:
> I think you should remember this error in `SortExec`, so if `initCompareFuncs` fails, every call of `Next` will return error, otherwise, the second call of `Next` will go to L266 directly?

`Next` is called by only one go routine, and once it returns error `Next` will not be called anymore. I think it's needless to remember it in `SortExec`.